contracts/MultiSigAdmin.sol

```diff
@@ -0,0 +1,667 @@
+    event ProposalExecuted(uint256 indexed id, address indexed executor);
+
+    /**
+     * @notice Ensure that the configuration for the given type of contract call
+     * is present
+     * @param targetContract    Address of the contract
+     * @param selector          Selector of the function in the contract
+     */
+    modifier configurationExists(address targetContract, bytes4 selector) {
+        require(
+            _types[targetContract][selector].config.minApprovals > 0,
+            "MultiSigAdmin: configuration does not exist"
+        );
+        _;
+    }
+
+    /**
+     * @notice Ensure that the caller is an approver for the given type of
+     * contract call
+     * @param targetContract    Address of the contract
+     * @param selector          Selector of the function in the contract
+     */
+    modifier onlyApprover(address targetContract, bytes4 selector) {
+        require(
+            _types[targetContract][selector].config.approvers.contains(
+                msg.sender
+            ),
+            "MultiSigAdmin: caller is not an approver"
+        );
+        _;
+    }
+
+    /**
+     * @notice Ensure that the caller is the proposer of a given proposal
+     * @param proposalId    Proposal ID
+     */
+    modifier onlyProposer(uint256 proposalId) {
+        require(
+            _proposals[proposalId].proposer == msg.sender,
+            "MultiSigAdmin: caller is not the proposer"
+        );
+        _;
+    }
+
+    /**
+     * @notice Ensure that the proposal is open
+     * @param proposalId    Proposal ID
+     */
+    modifier proposalIsOpen(uint256 proposalId) {
+        require(
+            _proposals[proposalId].state == ProposalState.Open,
+            "MultiSigAdmin: proposal is not open"
+        );
+        _;
+    }
+
+    /**
+     * @notice Ensure that the caller can approve a given proposal
+     * @param proposalId    Proposal ID
+     */
+    modifier onlyApproverForProposal(uint256 proposalId) {
+        Proposal storage proposal = _proposals[proposalId];
+        require(
+            _types[proposal.targetContract][proposal.selector]
+                .config
+                .approvers
+                .contains(msg.sender),
+            "MultiSigAdmin: caller is not an approver"
+        );
+        _;
+    }
+
+    constructor() public Administrable() {}
+
+    /**
+     * @notice Configure who can approve and how many approvals are required for
+     * a given type of contract call
+     * @dev minApprovals must be greater than zero.
+     * @param targetContract    Address of the contract
+     * @param selector          Selector of the function in the contract
+     * @param minApprovals      Minimum number of approvals required
+     * @param approvers         List of approvers' addresses
+     */
+    function configure(
+        address targetContract,
+        bytes4 selector,
+        uint256 minApprovals,
+        address[] calldata approvers
+    ) external onlyAdmin {
```

Comment:
Should this method invalidate any existing open proposals? Or leave them open, but reset the approvals? `_execute` only checks the number of approvals, but not that the approvals are in the current set of approvers.

This might let 1 entity execute a proposal if the approval addresses are periodically rotated without calling `removeConfiguration`. 
Soemthing like:
Alice, Bob, Carol have a 2 of 3 multisig setup.
Alice makes a proposal and approves it.
Addresses are rotated to new addresses controlled by Alice, Bob, Carol
Alice approves again from new address and executes, even though neither Bob nor Carol ever approved

Reply:
Yeah I think it makes sense for us to make it invalidate existing proposals.